targets/psa_switch/psa_switch.h

```diff
@@ -79,6 +83,28 @@ class PsaSwitch : public Switch {
   // by default, swapping is off
   explicit PsaSwitch(bool enable_swap = false);
 
```

Comment:
What I did here might seem hacky, but I did not want to interfere with the runtime interface directly since simple switch will also be affected. Instead I did an override of the read_counters and write_counters functions which would then call the new methods I wrote in context.cpp. This way, the interface for both psa and simple switch stays the same but each switch reads from their own implementation of counter.